web/src/stream_popover.js

```diff
@@ -362,6 +363,79 @@ export function build_move_topic_to_stream_popover(current_stream_id, topic_name
     });
 }
 
+export function build_topic_edit_popover(current_stream_id, topic_name) {
```

Comment:
Hmm, should this be called `build_topic_edit_popover`? The thing it builds isn't actually a popover, it's a modal. The code is just in `stream_popover.js` because it's launched from the popover.

Probably the right thing to do is to actually move the move topic modal launching functions to a separate file, `topic_edit_modals.js`, and name them appropriately, so that this file just retains the code for the popover itself, not these modals that can be launched from there. So we could leave that for a follow-up if you'd like.